dory/Hardware_targets/GAP8/Common/Utils/dory_dma.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * dory.h
+ * Alessio Burrello <alessio.burrello@unibo.it>
+ *
+ * Copyright (C) 2019-2020 University of Bologna
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _DORY_DMA_H
+#define _DORY_DMA_H
+
+#include "pmsis.h"
+
+#ifndef MCHAN_BASE_ADDR
+// FIXME: For GAP9, this must point to ARCHI_MCHAN_EXT_ADDR!!!
+// In PULP-SDK for Kraken, this is fixed.
+// GAP8 hardware to be tested...
+#define MCHAN_BASE_ADDR (ARCHI_MCHAN_DEMUX_ADDR)  // CLUSTER_MCHAN_ADDR
+#endif
+#define MCHAN_EVENT
+//#define MCHAN_POLLED
+#ifdef MCHAN_EVENT
+#define MCHAN_EVENT_BIT (ARCHI_CL_EVT_DMA0)  // 8
+#endif
+#include "mchan.h"
+
+
+#if   defined(MCHAN_POLLED)
+#define MCHAN_FLAGS    (MCHAN_CMD_FLAG_INCREMENTAL)
+#elif defined(MCHAN_EVENT)
+#define MCHAN_FLAGS    (MCHAN_CMD_FLAG_EVENT_ENABLE | MCHAN_CMD_FLAG_INCREMENTAL)
+#elif defined(MCHAN_INTERRUPT)
+#define MCHAN_FLAGS    (MCHAN_CMD_FLAG_INTERRUPT_ENABLE | MCHAN_CMD_FLAG_INCREMENTAL)
+#endif
+
+#define MCHAN_FLAGS_1D (MCHAN_FLAGS)
+#define MCHAN_FLAGS_2D (MCHAN_FLAGS | MCHAN_CMD_FLAG_2D_TRANSFER_EXTERNAL)
+
+#define MIN(a,b) ((a)<(b)?(a):(b))
```

Comment:
Could you move this to the `dory_dma.c` file and keep the header file as minimal as required?

Reply:
done!